substrate/frame/support/src/storage/types/nmap.rs

```diff
@@ -101,22 +103,54 @@ where
 	OnEmpty: Get<QueryKind::Query> + 'static,
 	MaxValues: Get<Option<u32>>,
 {
-	type Query = QueryKind::Query;
 	fn pallet_prefix() -> &'static [u8] {
 		Prefix::pallet_prefix().as_bytes()
 	}
 	fn storage_prefix() -> &'static [u8] {
 		Prefix::STORAGE_PREFIX.as_bytes()
 	}
-	fn prefix_hash() -> [u8; 32] {
+	pub fn prefix_hash() -> [u8; 32] {
 		Prefix::prefix_hash()
 	}
-	fn from_optional_value_to_query(v: Option<Value>) -> Self::Query {
+	fn from_optional_value_to_query(v: Option<Value>) -> QueryKind::Query {
 		QueryKind::from_optional_value_to_query(v)
 	}
-	fn from_query_to_optional_value(v: Self::Query) -> Option<Value> {
+	fn from_query_to_optional_value(v: QueryKind::Query) -> Option<Value> {
```

Comment:
maybe keep it public or deprecated

Reply:
it's still required in `storage::types`

Reply:
yes I mean some user could also call those functions, so we could keep it public or public and deprecated.

internally we can rename and have a private function.

Reply:
We can still make use of the deprecated funtions internally, provided the new funtions will still be the same as the ones deprecated?

Reply:
I'm not sure I understand.
But all in all, I think just keeping them public is good.